code/game/machinery/flasher.dm

```diff
@@ -23,6 +23,7 @@
 	density = 1
 
 /obj/machinery/flasher/New()
+	..() // ..() is EXTREMELY IMPORTANT, never forget to add it

```

Comment:
I spent a hour debugging this shit, and then suddenly understood that someone could forget ..() in New(), and that's why a machine isn't on machines list.